p/scripts/main.js

```diff
@@ -565,20 +565,25 @@ function init_column_categories() {
 				.append(template).find('button.confirm').removeAttr('disabled');
 		} else {
 			var id = $(this).closest('.item').attr('id').substr(2);
-
-			if ($(this).next('.dropdown-menu').css('display') === 'none'){
-				$(this).attr('href', '#dropdown-' + id);
-			} else {
-				$(this).attr('href', "#close");
-			}
 		}
 	});
 }
 
+var dropdownHref = null;
+
 function init_dropdown_handler() {
 	$('#global').not($('.dropdown')).click(function () {
 		window.location.hash = "close";
 	});
+	
+	$('.dropdown-toggle').click(function() {
```

Comment:
This adds many listeners, which I try to avoid, even though the extra resource consumption might not be so significant. I will try to check if it is possible to move it back to a single listener.

Reply:
I'll be most curious to see how you do it, if you can move it back to a single listener. I _think_ that my JavaScript has improved some since I've had occasion to do more with the project, but that's still just an improvement from blundering idiot to slightly-less-blundering idiot.

Reply:
I'd argue it's hard if not impossible to improve your JavaScript and DOM scripting by playing with jQuery, but that aside.

At a quick glance it looks like you're already adding effectively the same event listener a few lines higher (557 original, 543 post-diff), which might be what @Alkarex is getting at.

Reply:
@Frenzie, noted on jQuery. Just wanted to be clear that, poor as it may be, I am making an effort to improve. There's just a lot I don't know that I don't know, and I'm trying to chip away at at least that. As always, I appreciate your guidance.

Pointing out the effectively duplicate listener made it obvious to me; it's the same thing, just for a different scope. I _think_ that this is on the right track to taking care of that, but I didn't want to commit it in the event I was wrong.

https://github.com/pattems/FreshRSS/blob/80caae05a595a25471f54239d20b8aa69986b80e/p/scripts/main.js#L538-L589

Reply:
Not tested yet, but this looks good @pattems 👍 
Sorry for my short message. In jQuery parlance, what I wanted is called *delegated events* and there is an explanation of the advantages on https://api.jquery.com/on/